xarray/core/computation.py

```diff
@@ -664,8 +676,10 @@ def stack(objects, dim, new_coord):
 
     signature = kwargs.pop('signature', None)
     join = kwargs.pop('join', 'inner')
+    data_vars_join = kwargs.pop('data_vars_join', 'inner')
+    keep_attrs = kwargs.pop('keep_attrs', True)
     exclude_dims = kwargs.pop('exclude_dims', frozenset())
-    dataset_fill_value = kwargs.pop('dataset_fill_value', None)
+    dataset_fill_value = kwargs.pop('dataset_fill_value', np.nan)
```

Comment:
I would actually rather force `dataset_fill_value` to be set explicitly.

One option is to create a top level constant for the default:
```
_DEFAULT_FILL_VALUE = object()

...
dataset_fill_value = kwargs.pop('dataset_fill_value', _DEFAULT_FILL_VALUE)
```

Then, in `collect_dict_values` raise an error if you need to use the fill value and `fill_value is _DEFAULT_FILL_VALUE`.